### ISSUE TITLE
problème des résultats 

### ISSUE THREAD

**Issue by fedala56** [nabeel-oz/qlik-py-tools#105]
Bonjour, 
je me permets de vous écrire afin de vous exposer mon problème. 
1. j'ai pris votre application "Qonnections-ML Model(Completed App) et je l'ai appliqué sur mes données tous ça fonction bien , j'ai eu des résultats , et sur les même données j'ai fais une régression logistique sous python pour que je compare les résultats , mais malheureusement si je prends juste par exemple la matrice de confusion j'obtiens pas les même résultats y a une grande différence entre celle de python et celle de Qlik. j'aimerai bien comprendre pourquoi on obtient des résultats différents.
2.  Pour la feuille "Actual vs Predictions" : Pour on calcul les prédictions pour toute la data par sur le test-set et après on compare les résultats avec le test_set réel .
Cordialement 
Nedjmila

**Comment by nabeel-oz** [comment]
Hi @fedala56 ,

I don't speak French so I've used Google Translate on your post. I'm not sure if I understood the second point of your post correctly.

Assuming the model parameters are the same, the difference could be in the data split or how the confusion matrix is being calculated.

We are using K-fold cross validation in the sample app. The confusion matrix is built using results from the [sklearn.model_selection.cross_val_predict](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_predict.html) method. 

```
# Get cross validation predictions for the confusion matrix
y_pred = cross_val_predict(self.model.pipe, self.X_train, y_train, cv=self.model.cv, fit_params=fit_params)
```
In the app `cv` is set to 3. So this method splits the data into 3 different train and test sets, trains an estimator on each train set and returns predictions from each test set. Each sample ends up in the test set once.

So we can use the full training data to build the confusion matrix. I'm using the `sklearn.metrics.confusion_matrix` method.

To calculate metrics using a hold-out dataset you can use the `PyTools.sklearn_Calculate_Metrics(model_name, n_features)` method. This is described [here](https://github.com/nabeel-oz/qlik-py-tools/blob/master/docs/scikit-learn.md#training-and-testing-the-model).

**Comment by fedala56** [comment]
Bonjour
Je me permet de vous contacter une autre fois, pour un autre problème sur l’application « Sample-App-Forecasting-with-Keras » quand j’essaie de télécharger les données il m’affiche une erreur qui la suivante « il fait appel au modèle …v1.h5, mais quand je regarde le chemin d’enregistrement le modèle est enregstré …v1.joblib. Est-ce que vous pouvez m’aider à corriger cette erreur s’il vous plait. Vous trouverez aussi des captures d’ecran sur l’erreur et le chemin d’enregistrement
Cordialement
Nedjmila Fedala


[cid:image001.png@01D6A0B1.E9C92F20][cid:image003.jpg@01D6A0B1.E9C92F20]

**Comment by nabeel-oz** [comment]
Hi Nedjmila, the images failed to upload on GitHub so I can't see the screenshots. Can you please try to repost? Also, did you see any errors when reloading the app?

**Comment by fedala56** [comment]
Bonjour
Vous trouverez en pièce jointe et en forme PDF les captures d’écrans que j’ai envoyé la dernière fois , il m’affiche cette erreur la quand j’essaie de rechargé les données.
Bien Cordialement
Nedjmila Fedala

**Comment by nabeel-oz** [comment]
That didn't seem to work either. You'll have to login to GitHub to post the attachments as they do not seem to get uploaded when replying by email. 

https://github.com/nabeel-oz/qlik-py-tools/issues/105

**Comment by fedala56** [comment]
Bonjour 
voila le fichier que j'ai insérer l'erreur 

[erreur.pdf](https://github.com/nabeel-oz/qlik-py-tools/files/5437651/erreur.pdf)
Merci

**Comment by nabeel-oz** [comment]
Hi, the error from Qlik doesn't tell us enough. Can you look at the logs from the terminal where you are running the SSE? 

If the H5 file is missing there was an error while training or saving the Keras model. It may be an error in the installation of some required package like Keras or Tensorflow.
